### DISCUSSION TITLE
$rawQuery and nested types

### DISCUSSION THREAD

**Discussion by webnoob** [prisma/prisma#5316] [Q&A] [answered]
Consider I did something like:

```js
const model = await prisma.arcs.findMany({
  include: {
    book: true
  }
}

// can now use model.book
```
If I wanted to do the same in SQL but still leverage the types, how would I write the query. I've tried:
```sql
const model = await prisma.$rawQuery`select 
a.*,
book.*
from
arc as a
inner join book as book on book.id = a.bookId`

// Can't use model.book as it's undefined
```
but that doesn't work.

I'm going to try and analyze the running queries prisma ends up running but thought I would ask.

EDIT: If what I'm looking at is correct, Prisma actually runs multiple queries when we do an include. it doesn't work off the joins etc

**Comment by webnoob** [comment, marked as answer]
Doing this seems to have worked but I would have prefered some under the hood magic as this will be tedious for many includes (helper function time I think ...) 😄 
```ts
const books = await this.photonService.$queryRaw`select * from book where id in (${join(
  models.map(m => m.bookId)
)})`

models.forEach(arc => {
  arc.book = books.find(f => f.id === arc.bookId)
})
```

**Reply by ryands17** [reply]
@webnoob 👋 

Yes Prisma doesn't use joins so it would be difficult to mimic the behaviour. If you're using Postgres, the best I got to was by using `json` helpers that it provides:

```ts
const users = await prisma.$queryRaw`
  select json_agg(urs) from (
    select u.*,
    (select json_agg(pts) from
      (select po.* from "post" po where po.author_id = u.id) pts
    ) posts
    from "user" u
  ) urs
  `

console.log(users[0].json_agg)
```

I suppose this would be better than mapping it on the application side.

**Reply by webnoob** [reply]
I'm using mysql unfortunately. I'll have a look and see if something similar exists.

Thanks for the response.

**Reply by webnoob** [reply]
Might be hopeful on this: 
![image](https://user-images.githubusercontent.com/10404747/106119197-d2625300-614c-11eb-80ac-1ccf26214265.png)

In: https://dev.mysql.com/worklog/task/?id=7987

EDIT: This won't do it.

**Reply by ctilley83** [reply]
How did you populate your models array?

**Comment by kvenn** [comment]
I also rely on raw queries a fair amount. Would be great to have nesting support on raw.

**Reply by MorenoMdz** [reply]
Just got to this post, same here.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by IArnaut2** [comment]
I have this query:
```TS
const result = await prisma.$queryRawUnsafe<Listing[]>(`
  SELECT *
  FROM "public"."Listing"
  LEFT JOIN "public"."Apartment" ON "public"."Apartment".id = "public"."Listing"."apartmentId"
`);
console.log(result);
```
And all of the entries are NOT structured (NO nested apartments in listings).
```
[
  {
    id: 1,
    title: 'A',
    description: 'A',
    createdAt: 2025-06-01T16:15:35.934Z,
    updatedAt: 2025-06-01T16:15:35.934Z,
    apartmentId: 1,
    termsId: 1,
    location: 'Grbavica',
    floor: 0,
    area: 30,
    price: 30,
    numOfRooms: 0,
    state: 'Izvorno',
    heating: 'Gradsko',
    equipment: 'Namešten',
    buildingId: 1
  },
  {
    id: 2,
    title: 'B',
    description: 'B',
    createdAt: 2025-06-01T22:39:25.675Z,
    updatedAt: 2025-06-01T22:39:25.675Z,
    apartmentId: 2,
    termsId: 2,
    location: 'Grbavica',
    floor: 1,
    area: 30,
    price: 300,
    numOfRooms: 0,
    state: 'Izvorno',
    heating: 'Gradsko',
    equipment: 'Namešten',
    buildingId: 2
  },
  {
    id: 3,
    title: 'C',
    description: 'C',
    createdAt: 2025-06-02T22:15:09.277Z,
    updatedAt: 2025-06-02T22:15:09.277Z,
    apartmentId: 3,
    termsId: 3,
    location: 'Grbavica',
    floor: 0,
    area: 30,
    price: 30,
    numOfRooms: 0,
    state: 'Izvorno',
    heating: 'Gradsko',
    equipment: 'Namešten',
    buildingId: 3
  }
]
```

**Reply by IArnaut2** [reply]
And you expect me to manually map everything into nested entries and waste lines of code?
